src/lib/swapper/swappers/OsmosisSwapper/OsmosisSwapper.ts

```diff
@@ -310,13 +306,29 @@ export class OsmosisSwapper implements Swapper<ChainId> {
       ? 'uosmo'
       : atomOnOsmosisAssetId.split('/')[1].replace(/:/g, '/')
 
+    const maybeRateInfo = await getRateInfo(
+      sellAsset.symbol,
+      buyAsset.symbol,
+      sellAmountCryptoBaseUnit !== '0' ? sellAmountCryptoBaseUnit : '1',
```

Comment:
this line is setting a zero amount to 1 because it's aware of the inner working of `getPoolRateInfo` where the sell amount is being used as the denominator in the rate calculation which results in divide by zero. 

```
const buyAmountCryptoBaseUnit = buyAssetInitialPoolSize.minus(buyAssetFinalPoolSize).toFixed(0)
const rate = bnOrZero(buyAmountCryptoBaseUnit).dividedBy(sellAmount).toString()
```

This in theory shouldn't be an issue since a zero sell amount should result in a zero numerator also, but this conditional should really be handled inside `getPoolRateInfo`. Happy to get this in as-is but ideally this will be addressed in a follow up

Reply:
I'd be happy with `getPoolRateInfo` early exiting with sane values, or throwing since a zero sell amount is nonsense anyway

Reply:
Moved the logic to `getPoolRateInfo` in [d41fc99fc](https://github.com/shapeshift/web/pull/4905/commits/d41fc99fc8631e10fb47b170197d3a1b3ca69126) - throwing on 0 might break things silently, will give it a try as a separate PR